SCR/scr/tests/test_gpsummaryupdate.py

```diff
@@ -0,0 +1,110 @@
+import unittest
+from common.utilities import Utilities
+from lxml import etree
+from definitions import ROOT_DIR
+from pathlib import Path
+from scr.gpsummaryupdate import SummaryCareRecord
+
+
+class FullTest(unittest.TestCase):
+
+    summaryCareRecord = SummaryCareRecord()
+
+    def test_basic(self):
+        """
+        A basic test using the clean summary update from the spine tests
+        :return:
+        """
+        root = etree.parse(str(Path(ROOT_DIR) / 'scr/tests/test_xmls/cleanSummaryUpdate.xml'))
+        expected = etree.tostring(root)
+        render = self.summaryCareRecord.populate_template_with_file(
+            str(Path(ROOT_DIR) / 'scr/tests/hashes/hash16UK05.json'))
+        Utilities.assert_xml_equal(expected, render)
+
+    def test_extended_html(self):
+        """
+        Uses a larger set of Html for the human readable contents
+        :return:
+        """
+        root = etree.parse(str(Path(ROOT_DIR) / 'scr/tests/test_xmls/SummaryUpdateExtendedContents.xml'))
+        expected = etree.tostring(root)
+
+        file_path = str(Path(ROOT_DIR) / 'scr/tests/hashes/extendedHTMLhash.json')
+        render = self.summaryCareRecord.populate_template_with_file(file_path)
+
+        Utilities.assert_xml_equal(expected, render)
+
+    def test_empty_html(self):
+        """
+        A test for an empty human readable content value
+        """
+        root = etree.parse(str(Path(ROOT_DIR) / 'scr/tests/test_xmls/EmptyHtmlGpSummaryUpdate.xml'))
+        expected = etree.tostring(root)
+
+        file_path = str(Path(ROOT_DIR) / 'scr/tests/hashes/emptyHtmlHash.json')
+        render = self.summaryCareRecord.populate_template_with_file(file_path)
+
+        Utilities.assert_xml_equal(expected, render)
+
+    def test_empty_hash(self):
+        """
+        Tests the contents are empty when a completely blank hash is provided
+        """
+        root = etree.parse(str(Path(ROOT_DIR) / 'scr/tests/test_xmls/EmptyHash.xml'))
+        expected = etree.tostring(root)
```

Comment:
You convert tostring here and then pass this to your utility function `assert_xml_equal` which I think objectifies it and converts it tostring again. Is this step required if the Utility function is doing this step also?

Reply:
This is a good point, there isn't really any reason for objectifying it at the beginning of the test other than just to read the data from the file, I could change it to read the string from the file as normal I/O if you think that's better?

Reply:
Yeah if you don't really need the objectifying of the file then might cut out a few unnecessary steps. Might not need the Utility function if both files are read in as strings that can be compared 🤷‍♂️

Reply:
Yeah maybe, though I think it's the same number of lines to read the file normally. I think I'll still need the utility method as a simple string comparison was having issues with formatting and new lines  when I first tried it

Reply:
ah yeah that would makes sense.